### PR TITLE
feat: Allow result text in Resume API/CLI. Fixes #3132

### LINKED ISSUE #3132: Be able to set result text when resuming a workflow
# Summary

Ability to set result text against a node when doing a resume via API or CLI

# Motivation

If the result of a resume may affect execution of future nodes.

# Proposal

#3133 

---

**Message from the maintainers**:

If you wish to see this enhancement implemented please add a 👍 reaction to this issue! We often sort issues this way to know what to prioritize.

### PR DESCRIPTION
**Pull request by mark9white**
Checklist:

* [X] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [X] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [X] I've signed the CLA.
* [X] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [X] My builds are green. Try syncing with master if they are not. 
* [X] My organization is added to [USERS.md](https://github.com/argoproj/argo/blob/master/USERS.md).

### CODE REVIEW

**Comment by mark9white** [comment]
@alexec checking if you're ok with this in principle, as we'd like to build on top of it internally.

File: server/workflow/workflow_server.go [hunk 1]
```diff
@@ -229,7 +229,7 @@ func (s *workflowServer) ResumeWorkflow(ctx context.Context, req *workflowpkg.Wo
 		return nil, err
 	}
 
-	err = util.ResumeWorkflow(wfClient.ArgoprojV1alpha1().Workflows(req.Namespace), s.hydrator, req.Name, req.NodeFieldSelector)
+	err = util.ResumeWorkflow(wfClient.ArgoprojV1alpha1().Workflows(req.Namespace), s.hydrator, req.Name, req.NodeFieldSelector, req.Result)
```
Comment (simster7) [inline server/workflow/workflow_server.go:232]: Perhaps we need to a `type resumeOpts struct { ... }`?

Comment (mark9white) [reply server/workflow/workflow_server.go]: I've applied this feedback.

File: workflow/util/util.go [hunk 2]
```diff
@@ -418,6 +430,12 @@ func updateWorkflowNodeByKey(wfIf v1alpha1.WorkflowInterface, hydrator hydrator.
 					if len(message) > 0 {
 						node.Message = message
 					}
+					if len(result) > 0 {
+						if node.Outputs == nil {
+							node.Outputs = &wfv1.Outputs{}
+						}
+						node.Outputs.Result = &result
```
Comment (dtaniwaki) [inline workflow/util/util.go:437]: 👍

**Comment by mark9white** [comment]
Sure, I'll do it next week.

**Comment by mark9white** [comment]
@simster7 Hey, just checking where this is at? Shall I rebase for merge?

**Comment by mark9white** [comment]
(oops, closed by accident)

**Comment by simster7** [comment]
Hi @mark9white, sorry for the delay. We decided to support this use case, however in a slightly different manner. Mainly we will use output parameters instead of output results for this. I'll work on a PR and ask for your review when it's done soon

**Comment by mark9white** [comment]
@simster7 thanks for the update - is it #3314 ?

**Comment by simster7** [comment]
> @simster7 thanks for the update - is it #3314 ?

Yes, sorry it's been a bit delayed

**Comment by simster7** [comment]
@mark9white, it is actually https://github.com/argoproj/argo/pull/3277 (not #3314), sorry for the confusion

**Comment by alexec** [comment]
Superseded.

**Pull request closed without merging**